cmd/tiller/tiller.go

```diff
@@ -157,12 +157,16 @@ func start() {
 			logger.Fatalf("Could not create server TLS configuration: %v", err)
 		}
 		opts = append(opts, grpc.Creds(credentials.NewTLS(cfg)))
-		opts = append(opts, grpc.KeepaliveParams(keepalive.ServerParameters{
-			MaxConnectionIdle: 10 * time.Minute,
-			// If needed, we can configure the max connection age
-		}))
 	}
 
+	opts = append(opts, grpc.KeepaliveParams(keepalive.ServerParameters{
+		MaxConnectionIdle: 10 * time.Minute,
+		// If needed, we can configure the max connection age
+	}))
+	opts = append(opts, grpc.KeepaliveEnforcementPolicy(keepalive.EnforcementPolicy{
+		MinTime: time.Duration(20) * time.Second, // For compatibility with the client keepalive.ClientParameters
```

Comment:
20 seconds is the default in client and server https://godoc.org/google.golang.org/grpc/keepalive#ClientParameters

Reply:
For the timeout yes, but the default enforcement policy is 5 minutes :)

https://godoc.org/google.golang.org/grpc/keepalive#EnforcementPolicy

Reply:
doh, good point. Need more ☕️